### PR TITLE
Add class library System.IO.FileSystem

### PR DESCRIPTION
**Pull request by NiciAndres**
## Description
Add new System.IO.FileSystem Library

## Motivation and Context
Make this library implemented by us available for everyone on nF.

## How Has This Been Tested?
It was tested in our intern project. Not yet tested as system library, because WIP

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by piwi1263** [comment]
Hi Nicholas, can it be that for the ORGPAL_THREE board you overlooked someting ?

image

**Comment by josesimoes** [comment]
@NiciAndres the CMake that you have is missing some entries for the path hints.
You really have to update you VS extension to have this fixed, along with other formatting bits that have been corrected there...

File: CMakeLists.txt [hunk 1]
```diff
@@ -570,7 +576,8 @@ if( API_nanoFramework.Devices.OneWire OR
     API_Windows.Devices.SerialCommunication OR
     API_Windows.Networking.Sockets OR
     API_nanoFramework.Devices.Can OR
-    API_Windows.Storage)
+    API_Windows.Storage OR
+    API._System.IO.FileSystem)
```
Comment (josesimoes) [inline CMakeLists.txt:580]: Wrong char here: `API._` that dot is wrong.

Tried to fix this, but I dont' have write permission to the fork you're using.

File: targets/CMSIS-OS/ChibiOS/ORGPAL_PALTHREE/cmake-variants.json [hunk 2]
```diff
@@ -49,6 +49,7 @@
           "NF_SECURITY_MBEDTLS": "ON",
           "SWO_OUTPUT": "OFF",
           "NF_BUILD_RTM": "OFF",
+          "API_System.IO:FileSystem": "ON",
```
Comment (josesimoes) [inline targets/CMSIS-OS/ChibiOS/ORGPAL_PALTHREE/cmake-variants.json:52]: Wrong ":" there as already pointed by @piwi1263

**Review by josesimoes — COMMENTED** [review]
Also the copyright headers need to be changed to match the latest version in use.

File: CMakeLists.txt (hunk 1 above)
Comment (NiciAndres) [reply CMakeLists.txt]: Thanks for the hint.

Comment (josesimoes) [reply CMakeLists.txt]: I've added the header.

**Pull request merged**